[BUG] format_iso_utc promove datetime naive usando fuso local em vez de UTC

## Descrição do Bug

format_iso_utc (em deile/orchestration/pipeline/_time_utils.py) promove um datetime naive (sem tzinfo) para UTC usando o fuso local do
sistema em vez de interpretá-lo como UTC. Isso faz o teste test_format_iso_utc_promotes_naive_to_utc (em
deile/tests/orchestration/pipeline/test_time_utils.py) falhar em qualquer máquina cujo fuso ≠ UTC (ex.: BRT, UTC−3). O próprio nome do
teste — promotes_naive_to_utc — indica que o contrato é "promover" (anexar UTC), não "converter do fuso local".

## Severidade

- Crítica — perda de dados, problema de segurança ou incapacidade total de executar o DEILE
- Maior — uma funcionalidade central está quebrada ou produz resultado errado; sem contorno
- Menor — comportamento degradado ou falha em caso de borda; existe contorno
- Cosmética — problema de exibição, formatação ou texto sem impacto funcional

## Passos para Reproduzir

1. Configurar o fuso do sistema para America/Sao_Paulo (UTC−3) — ou qualquer fuso ≠ UTC.
2. Rodar python3 -m pytest deile/tests/orchestration/pipeline/test_time_utils.py -q.
3. Observar a falha de asserção (offset de exatamente 3h).

## Comportamento Esperado

Um datetime naive deve ser interpretado como UTC (apenas anexar tzinfo=timezone.utc), produzindo 2026-01-02T03:04:05Z.

## Comportamento Atual

Em máquina UTC−3 a saída vem 3 horas adiantada (06:04:05Z), porque a função aparenta aplicar dt.astimezone(timezone.utc) sobre um datetime
naive — e astimezone em naive assume o fuso local do SO. O correto para naive é dt.replace(tzinfo=timezone.utc); só datetimes aware devem
passar por astimezone(timezone.utc).

## Saída de Erro

FAILED deile/tests/orchestration/pipeline/test_time_utils.py::test_format_iso_utc_promotes_naive_to_utc
AssertionError: assert '2026-01-02T06:04:05Z' == '2026-01-02T03:04:05Z'

## Ambiente

- SO: macOS (Darwin 25.4.0), fuso America/Sao_Paulo (UTC−3)
- Python: 3.11 (cluster) / 3.14 (suíte local)
- Versão / commit do DEILE: ramo main (ver deile/__version__.py)
- Provider + modelo em uso: N/A (falha em teste unitário, sem chamada a LLM)

## Contexto Adicional

Correção proposta: em format_iso_utc, ramificar pelo tzinfo — naive (dt.tzinfo is None) → dt.replace(tzinfo=timezone.utc) (promover);
aware → dt.astimezone(timezone.utc) (converter). Validar/fixar no teste qual é o contrato oficial. Quem implementar deve ler a função
antes de fechar a causa-raiz — a hipótese (astimezone sobre naive) vem do sintoma (offset = exatamente o fuso local), mas precisa ser
confirmada no código. É a única falha recorrente da suíte (3061 passam); não relacionada às mudanças de persona reviewer / fix de menção
desta branch.

--- a/deile/orchestration/pipeline/_time_utils.py
+++ b/deile/orchestration/pipeline/_time_utils.py
@@ -22,7 +22,11 @@ def format_iso_utc(dt: Optional[datetime]) -> Optional[str]:
     """Format ``dt`` as ``YYYY-MM-DDTHH:MM:SSZ`` (UTC). ``None`` passes through."""
     if dt is None:
         return None
-    return dt.astimezone(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+    if dt.tzinfo is None:
+        dt = dt.replace(tzinfo=timezone.utc)
+    else:
+        dt = dt.astimezone(timezone.utc)
+    return dt.strftime("%Y-%m-%dT%H:%M:%SZ")
 
 
 def parse_iso_utc(value) -> Optional[datetime]:

--- a/deile/tests/orchestration/pipeline/test_time_utils.py
+++ b/deile/tests/orchestration/pipeline/test_time_utils.py
@@ -2,7 +2,7 @@
 
 from __future__ import annotations
 
-from datetime import datetime, timezone
+from datetime import datetime, timedelta, timezone
 
 import pytest
 
@@ -70,3 +70,11 @@ def test_parse_iso_utc_rejects_empty_string():
 def test_parse_iso_utc_rejects_bare_z():
     with pytest.raises(ValueError, match="invalid ISO datetime"):
         parse_iso_utc("Z")
+
+
+def test_format_iso_utc_converts_aware_to_utc():
+    """Aware datetime in a non-UTC zone must be converted to UTC."""
+    brt = timezone(offset=timedelta(hours=-3))
+    dt = datetime(2026, 1, 2, 6, 4, 5, tzinfo=brt)  # 06:04 BRT = 09:04 UTC
+    s = format_iso_utc(dt)
+    assert s == "2026-01-02T09:04:05Z"

